src/seismometer/controls/explore.py

```diff
@@ -0,0 +1,907 @@
+        plot_function : Callable[..., Any]
+            callable to generate this plot: Expected to have the following signature:
+
+            def plot_function(
+                cohorts: dict[str,tuple[Any]]
+                target: str,
+                score: str,
+                thresholds: tuple[float],
+                *, per_context: bool) -> Any
+        """
+        from seismometer.seismogram import Seismogram
+
+        sg = Seismogram()
+        thresholds = {f"Threshold {k}": v for k, v in enumerate(sorted(sg.thresholds, reverse=True), 1)}
+        super().__init__(
+            title=title,
+            option_widget=ModelOptionsAndCohortsWidget(
+                sg.available_cohort_groups, sg.target_cols, sg.output_list, thresholds, per_context=False
+            ),
+            plot_function=plot_function,
+        )
+
+    def generate_plot_args(self) -> tuple[tuple, dict]:
+        """Generates the plot arguments for the model evaluation plot"""
+        args = (
+            self.option_widget.cohorts,
+            self.option_widget.target,
+            self.option_widget.score,
+            list(self.option_widget.thresholds.values()),
+        )
+        kwargs = {
+            "per_context": self.option_widget.group_scores,
+        }
+        return args, kwargs
+
+
+class ExplorationCohortSubclassEvaluationWidget(ExplorationWidget):
+    """
+    A widget for exploring the model performance based on the subgroups of a cohort column.
+    """
+
+    def __init__(
+        self,
+        title: str,
+        plot_function: Callable[..., Any],
+        *,
+        ignore_grouping: bool = False,
+        theshold_handling: Literal["all", "max", "min", None] = "all",
+    ):
+        """
+        Exploration widget for model evaluation, showing a plot for a given target,
+        score, threshold, broken down across labels in a cohort column.
+
+        Parameters
+        ----------
+        title : str
+            title of the control
+        plot_function : Callable[..., Any]
+            callable to generate this plot: Expected to have the following signature:
+
+            def plot_function(
+                cohorts_col: str
+                cohort_subgroups: tuple[Any]
+                target: str,
+                score: str,
+                thresholds: tuple[float],
+                *, per_context: bool) -> Any
+        """
+        from seismometer.seismogram import Seismogram
+
+        sg = Seismogram()
+        match theshold_handling:
+            case "all":
+                self.thresholds = {f"Threshold {k}": v for k, v in enumerate(sorted(sg.thresholds, reverse=True), 1)}
+            case "max":
+                self.thresholds = {"Threshold": max(sg.thresholds)}
+            case "min":
+                self.thresholds = {"Threshold": min(sg.thresholds)}
+            case _:
+                self.thresholds = None
+
+        self.threshold_hanlding = theshold_handling
```

Comment:
```suggestion
        self.threshold_handling = threshold_handling
```
handling (on left), threshold (on right)  
I think this was also copied into some other classes

Reply:
FINC